Packs/AWS-GuardDuty/Integrations/AWSGuardDutyEventCollector/AWSGuardDutyEventCollector.py

```diff
@@ -0,0 +1,263 @@
+
+    Args:
+        events (list): Events received from AWS python SDK with datetime in certain fields.
+
+    Returns:
+        events (list): Events with dates as strings only.
+    """
+    for event in events:
+        if event.get('Resource'):
+            resource = event.get('Resource', {})
+            service = event.get('Service', {})
+            if resource.get('S3BucketDetails'):
+                s3bucket_details = resource.get('S3BucketDetails')
+                if s3bucket_details and type(s3bucket_details) == list:
+                    for s3buckt_detail in s3bucket_details:
+                        if type(s3buckt_detail.get('CreatedAt')) == datetime:
+                            s3buckt_detail['CreatedAt'] = s3buckt_detail['CreatedAt'].__str__()
+            if type(resource.get('EksClusterDetails', {}).get('CreatedAt')) == datetime:
+                resource['EksClusterDetails']['CreatedAt'] = resource['EksClusterDetails']['CreatedAt'].__str__()
+            if type(resource.get('EcsClusterDetails', {}).get('TaskDetails', {}).get('TaskCreatedAt')) == datetime:
+                resource['EcsClusterDetails']['TaskDetails']['TaskCreatedAt'] = \
+                    resource['EcsClusterDetails']['TaskDetails']['TaskCreatedAt'].__str__()
+            if type(resource.get('EcsClusterDetails', {}).get('TaskDetails', {}).get('StartedAt')) == datetime:
+                resource['EcsClusterDetails']['TaskDetails']['StartedAt'] = \
+                    resource['EcsClusterDetails']['TaskDetails']['StartedAt'].__str__()
+            if type(service.get('EbsVolumeScanDetails', {}).get('ScanStartedAt')) == datetime:
+                service['EbsVolumeScanDetails']['ScanStartedAt'] = \
+                    service['EbsVolumeScanDetails']['ScanStartedAt'].__str__()
+            if type(service.get('EbsVolumeScanDetails', {}).get('ScanCompletedAt')) == datetime:
+                service['EbsVolumeScanDetails']['ScanCompletedAt'] = \
+                    service['EbsVolumeScanDetails']['ScanCompletedAt'].__str__()
+
+    return events
+
+
+def get_events(aws_client: boto3.client, collect_from: dict, collect_from_default: Optional[datetime], last_ids: dict,
+               severity: str, limit: int = MAX_RESULTS, detectors_num: int = MAX_RESULTS,
+               max_ids_per_req: int = MAX_IDS_PER_REQ) -> Tuple[list, dict, dict]:
+    """Get events from AWSGuardDuty.
+
+    Args:
+        aws_client: AWSClient session to get events from.
+        collect_from: Dict of {detector_id: datestring to start collecting from}, used when fetching.
+        collect_from_default: datetime to start collecting from if detector id is not found in collect_from keys.
+        last_ids: Dict of {detector_id: last fetched id}, used to avoid duplicates.
+        severity: The minimum severity to start fetching from. (inclusive)
+        limit: The maximum number of events to fetch.
+        detectors_num: The maximum number of detectors to fetch.
```

Comment:
<img width="760" alt="Screen Shot 2022-09-11 at 17 32 29" src="https://user-images.githubusercontent.com/72339665/189533141-54e747f2-e737-4c3a-a982-ae12218bb59a.png">

detectors_num is used as an upper bound on MaxResults parameter, since MAX_RESULTS is the maximum value the api request accepts,  detectors_num < MAX_RESULTS always.
In reality, the number of detectors reachable for a user is usually 1 (as I understand it), so having more than MAX_RESULTS=50 and not being able to list all of them due to the API restrictions is an edge case. (I believe)

Reply:
https://docs.aws.amazon.com/guardduty/latest/APIReference/API_ListDetectors.html